CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -40,7 +40,8 @@ public enum ChildKind {
   /// The child always contains a node of the given `kind`.
   case node(kind: SyntaxNodeKind)
   /// The child always contains a node that matches one of the `choices`.
-  case nodeChoices(choices: [Child])
+  case nodeChoices(choices: [Child], childHistory: Child.History = [])
+  // FIXME: We don't appear to have ever generated compatibility layers for children of node choices!
```

Comment:
`nodeChoices` still doesn't have compatibility layers. I think largely because Swift can't alias cases—you can use a deprecated static method to construct the replacement case, but not to pattern-match it.

Reply:
Instead of having a `FIXME` for it (which is very easy to miss), could we raise a `fatalError` if you try to add a compatibility layer of node choices? That way you can’t miss it if you try to add one.

Reply:
So, there actually *are* existing cases where people have specified a `deprecatedName:` for a child of a `ChildKind.nodeChoices`; this information was simply ignored until now. Rather than deleting it, I thought it'd be better to preserve the info in the node definition files, but if we'd rather just delete it (which—again—would be an NFC change), I can remove the `childHistory` value here entirely.

Reply:
I haven't done anything with this yet, but I think I've addressed everything else you suggested.